library/core/src/androidMain/kotlin/com/attafitamim/krop/core/images/ImageStreamSrc.android.kt

```diff
@@ -85,8 +93,75 @@ fun bitmapFactoryOptions(sampleSize: Int) = BitmapFactory.Options().apply {
     inSampleSize = sampleSize
 }
 
-fun InputStream.getImageSize(): IntSize {
+suspend fun ImageStream.getOrientation(): Int {
+    val exif = tryUse(::ExifInterface) ?: return ExifInterface.ORIENTATION_UNDEFINED
+    return exif.getAttributeInt(
+        ExifInterface.TAG_ORIENTATION,
+        ExifInterface.ORIENTATION_UNDEFINED
+    )
+}
+
+suspend fun ImageStream.getImageSize(): IntSize? = tryUse { stream ->
     val options = BitmapFactory.Options().apply { inJustDecodeBounds = true }
-    BitmapFactory.decodeStream(this, null, options)
-    return IntSize(options.outWidth, options.outHeight)
+    BitmapFactory.decodeStream(stream, null, options)
+    options
+}?.let { options ->
+    val orientation = getOrientation()
+    val actualSize = if (isImageSizeFlipped(orientation)) {
+        IntSize(options.outHeight, options.outWidth)
+    } else {
+        IntSize(options.outWidth, options.outHeight)
+    }
+
+    actualSize
+}
+
+fun isImageSizeFlipped(orientation: Int): Boolean =
+    when (orientation) {
+        ExifInterface.ORIENTATION_ROTATE_90,
+        ExifInterface.ORIENTATION_ROTATE_270,
+        ExifInterface.ORIENTATION_TRANSPOSE,
+        ExifInterface.ORIENTATION_TRANSVERSE -> true
+        else -> false
+    }
+
+fun Bitmap.ensureCorrectOrientation(orientation: Int): Bitmap {
+    val matrix = Matrix().apply {
+        when (orientation) {
+            ExifInterface.ORIENTATION_NORMAL -> return this@ensureCorrectOrientation
+            ExifInterface.ORIENTATION_FLIP_HORIZONTAL -> setScale(-DEFAULT_SCALE, DEFAULT_SCALE)
+            ExifInterface.ORIENTATION_ROTATE_180 -> setRotate(VERTICAL_ROTATION_ANGLE)
+            ExifInterface.ORIENTATION_FLIP_VERTICAL -> {
+                setRotate(VERTICAL_ROTATION_ANGLE)
+                postScale(-DEFAULT_SCALE, DEFAULT_SCALE)
+            }
```

Comment:
This can be simplified:

```suggestion
            ExifInterface.ORIENTATION_FLIP_VERTICAL -> postScale(DEFAULT_SCALE, -DEFAULT_SCALE)
```